esx_service/vmdk_ops.py

```diff
@@ -1810,18 +1855,30 @@ def handleVmciRequests(port):
 
         client_socket = c # Bind to avoid race conditions.
 
+        # Check the flag - if set, fail new incoming requests
+        if stopEvent.isSet():
+            svc_stop_err = 'Service is being stopped: operation declined!'
+            logging.warning(svc_stop_err)
+            send_vmci_reply(client_socket, err(svc_stop_err))
+            continue
+
         if not get_si():
             svc_connect_err = 'Service is presently unavailable, ensure the ESXi Host Agent is running on this host'
             logging.warning(svc_connect_err)
             send_vmci_reply(client_socket, err(svc_connect_err))
             continue
 
+        # Increase the in-flight ops counter
+        opsCounter.incr()
+
         # Fire a thread to execute the request
         threadutils.start_new_thread(
             target=execRequestThread,
             args=(client_socket, cartel.value, txt.value))
 
-    vmci_release_listening_socket() # close listening socket when the loop is over
+    # Close listening socket when the loop is over
+    logging.info("Closing VMCI listening socket...")
+    vmci_release_listening_socket()
```

Comment:
wait , why do you want to close the listening socket ? Will it be reopened on the next cycle ? where ?

Reply:
This is existing code, I just added a log:) I suppose it's just for releasing the socket resource.

Reply:
I see. Ok, I'll assume.ots.corect then :-)

Reply:
Sorry about the typos, using phone....